### ISSUE TITLE
Fails to allocate memory

### ISSUE THREAD

**Issue by sjug** [stressapptest/stressapptest#46]
Unable to start test

`Linux ryzen 4.10.10-1-ARCH #1 SMP PREEMPT Wed Apr 12 18:50:28 CEST 2017 x86_64 GNU/Linux`

```
[sjug@ryzen stressapptest]$ stressapptest -W
2017/04/17-20:18:16(EDT) Log: Commandline - stressapptest -W
2017/04/17-20:18:16(EDT) Stats: SAT revision 1.0.7_autoconf, 64 bit binary
2017/04/17-20:18:16(EDT) Log: sjug @ ryzen on Mon Apr 17 20:17:29 EDT 2017 from open source release
2017/04/17-20:18:16(EDT) Log: 1 nodes, 16 cpus.
2017/04/17-20:18:16(EDT) Log: Defaulting to 16 copy threads
2017/04/17-20:18:16(EDT) Log: Total 16043 MB. Free 11838 MB. Hugepages 0 MB. Targeting 15049 MB (93%)
2017/04/17-20:18:16(EDT) Log: Prefer plain malloc memory allocation.
2017/04/17-20:18:16(EDT) Process Error: memalign returned 0
2017/04/17-20:18:16(EDT) Process Error: failed to allocate memory
2017/04/17-20:18:16(EDT) Process Error: Sat::Initialize() failed
2017/04/17-20:18:16(EDT)
2017/04/17-20:18:16(EDT) Status: FAIL - test encountered procedural errors
2017/04/17-20:18:16(EDT)
2017/04/17-20:18:16(EDT) Process Error: Fatal issue encountered. See above logs for details.
```

**Comment by nickjsanders** [comment]
In the log it indicates you only have 11G free memory, but stressapptest is
trying to allocate 15G. You can either free some memory by stopping other
processes, or specify -M 10000 which will allocate a smaller amount of
memory.

On Apr 17, 2017 5:20 PM, "Sebastian Jug" <notifications@github.com> wrote:

Unable to start test

Linux ryzen 4.10.10-1-ARCH #1 SMP PREEMPT Wed Apr 12 18:50:28 CEST 2017
x86_64 GNU/Linux

[sjug@ryzen stressapptest]$ stressapptest -W
2017/04/17-20:18:16(EDT) Log: Commandline - stressapptest -W
2017/04/17-20:18:16(EDT) Stats: SAT revision 1.0.7_autoconf, 64 bit binary
2017/04/17-20:18:16(EDT) Log: sjug @ ryzen on Mon Apr 17 20:17:29 EDT
2017 from open source release
2017/04/17-20:18:16(EDT) Log: 1 nodes, 16 cpus.
2017/04/17-20:18:16(EDT) Log: Defaulting to 16 copy threads
2017/04/17-20:18:16(EDT) Log: Total 16043 MB. Free 11838 MB. Hugepages
0 MB. Targeting 15049 MB (93%)
2017/04/17-20:18:16(EDT) Log: Prefer plain malloc memory allocation.
2017/04/17-20:18:16(EDT) Process Error: memalign returned 0
2017/04/17-20:18:16(EDT) Process Error: failed to allocate memory
2017/04/17-20:18:16(EDT) Process Error: Sat::Initialize() failed
2017/04/17-20:18:16(EDT)
2017/04/17-20:18:16(EDT) Status: FAIL - test encountered procedural errors
2017/04/17-20:18:16(EDT)
2017/04/17-20:18:16(EDT) Process Error: Fatal issue encountered. See
above logs for details.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/stressapptest/stressapptest/issues/46>, or mute the
thread

**Comment by sjug** [comment]
Yes I do get that. The question is why is the expected behavior to allocate more memory than is free?

**Comment by nickjsanders** [comment]
Linux will typically evict buffer cache and other noncritical memory as
needed, so the amount of currently free memory is not equal to the amount
of allocatable memory, and unfortunately the amount of allocatable memory
can't reliably be calculated in advance. On some configs it's possible that
buffer cache will fill all available memory, so free memory may always be
near zero in that situation.

Generally the stressapptest usecase doesn't include running other memory
intensive jobs in parallel, so it's unusual to have another process holding
onto 4G in the background. But, if you do have this situation, just
explicitly specify the amount of memory you'd like to allocate to
stressapptest.

**Comment by DaveyDoodle** [comment]
> Generally the stressapptest usecase doesn't include running other memory intensive jobs in parallel, so it's unusual to have another process holding onto 4G in the background.

You mean like... booting a Windows 10 PC that is having numerous BSODs from a Linux Mint live DVD, running diagnostics, installing stressapptest, and having to come here to figure out why it won't run? Sounds like a completely plausible usecase to me.
